ariadne/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -514,25 +515,18 @@ deriveBip44KeyPair ::
     -> HdAddressChain
     -> HdAddressIx
     -> Maybe (Address, EncryptedSecretKey)
-deriveBip44KeyPair era pp sk hdAccountIx hdAddressChain hdAddressIx =
-    derivePathKeyPair era pp sk $
-        encodeBip44DerivationPath $
-            Bip44DerivationPath hdAccountIx hdAddressChain hdAddressIx
-
--- ^ This function derives a key (and an address) following an arbitrary derivation path.
-derivePathKeyPair ::
-       IsBootstrapEraAddr
-    -> PassPhrase
-    -> EncryptedSecretKey
-    -> [Word32]
-    -> Maybe (Address, EncryptedSecretKey)
-derivePathKeyPair era pp sk derPath = do
-    addrKey <- mAddrKey
-    let hdPP = deriveHDPassphrase $ encToPublic sk
-        hdPayload = packHDAddressAttr hdPP derPath
-    return (makePubKeyHdwAddress era hdPayload (encToPublic addrKey), addrKey)
+deriveBip44KeyPair era pp rootSK hdAccountIx hdAddressChain hdAddressIx =
+    toPair <$>
+    deriveHDSecretKeyByPath (ShouldCheckPassphrase True) pp rootSK derPath
   where
-    mAddrKey = foldM
-        (\sk' (check, idx) -> deriveHDSecretKey (ShouldCheckPassphrase check) pp sk' idx)
-        sk
-        (zip (True:repeat False) derPath) -- check passphrase only once
+    bip44DerPath = Bip44DerivationPath hdAccountIx hdAddressChain hdAddressIx
```

Comment:
Maybe we should pack the 3 parameters of this function into one `Bip44DerivationPath`, so that you don't have to write this?

Reply:
Makes sense, will do.